### PR TITLE
Check for server readiness before starting ssl stress test

### LINKED ISSUE #94222: [SslStress] SocketException (111): Connection refused
This exception started to pop up quite often on CI in runs against `main` since 2023-10-14. To list a few runs:
- 3x in [report 10/14-19](https://github.com/dotnet/runtime/issues/42211#issuecomment-1770972051)
    - 2x in run [20231014.1](https://dev.azure.com/dnceng-public/public/_build/results?buildId=438734&view=logs&j=22fea640-1099-5f32-ec5d-316ad83f359a&t=f4d03b33-e265-5206-5dc1-b33aeb754840&l=13398)
    - 1x in run [20231018.1](https://dev.azure.com/dnceng-public/public/_build/results?buildId=442616&view=logs&j=22fea640-1099-5f32-ec5d-316ad83f359a&t=f4d03b33-e265-5206-5dc1-b33aeb754840&l=11383)
- 5x in [report 10/20-31](https://github.com/dotnet/runtime/issues/42211#issuecomment-1787544999)
    - https://dev.azure.com/dnceng-public/public/_build/results?buildId=445570&view=logs&j=22fea640-1099-5f32-ec5d-316ad83f359a&t=f4d03b33-e265-5206-5dc1-b33aeb754840
    - https://dev.azure.com/dnceng-public/public/_build/results?buildId=447860&view=logs&j=22fea640-1099-5f32-ec5d-316ad83f359a&t=f4d03b33-e265-5206-5dc1-b33aeb754840
    - https://dev.azure.com/dnceng-public/public/_build/results?buildId=451951&view=logs&j=22fea640-1099-5f32-ec5d-316ad83f359a&t=f4d03b33-e265-5206-5dc1-b33aeb754840

Considering the frequency, I think we should take a look within the 9.0 timeframe.

```
System.Net.Sockets.SocketException (111): Connection refused
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs:line 1378
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs:line 1294
   at System.Threading.Tasks.ValueTask.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/ValueTask.cs:line 258
--- End of stack trace from previous location ---
   at System.Net.Sockets.TcpClient.CompleteConnectAsync(Task task) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs:line 193
   at SslStress.SslClientBase.<StartCore>g__RunSingleWorker|15_1(Int32 workerId, Random random) in /app/System.Net.Security/tests/StressTests/SslStress/SslClientBase.cs:line 116
```

### PR DESCRIPTION
**Pull request by rzikm**
Fixes #94222.

I copied the approach from http stress, it seems to work well enough over there.

### CODE REVIEW

**Review by MihaZupan — APPROVED** [review]

File: src/libraries/System.Net.Security/tests/StressTests/SslStress/StressOperations.cs [hunk 1]
```diff
@@ -194,6 +194,34 @@ public sealed class StressClient : SslClientBase
...
+            Console.WriteLine($"Connected successfully.");
+
+            async Task TestConnection(int maxRetries)
+            {
+                for (int remainingRetries = maxRetries; ; remainingRetries--)
+                {
+                    try
+                    {
+                        using var client = new TcpClient();
+                        await client.ConnectAsync(_config.ServerEndpoint.Address, _config.ServerEndpoint.Port);
```
Comment (MihaZupan) [inline src/libraries/System.Net.Security/tests/StressTests/SslStress/StressOperations.cs:214]:
```suggestion
                        await client.ConnectAsync(_config.ServerEndpoint.Address, _config.ServerEndpoint.Port);
                        return;
```

**Review by antonfirsov — APPROVED** [review]
Looks good, thanks!

**Pull request merged**